apps/dashboard/src/components/number-input-with-select.tsx

```diff
@@ -0,0 +1,104 @@
+import { cn } from '@/utils/ui';
+import { FormControl, FormField, FormItem, FormMessagePure } from '@/components/primitives/form/form';
+import { Input } from '@/components/primitives/input';
+import { InputFieldPure } from '@/components/primitives/input';
+import { Select, SelectContent, SelectItem, SelectTrigger, SelectValue } from '@/components/primitives/select';
+import { useFormContext } from 'react-hook-form';
+import { useMemo } from 'react';
+
+type InputWithSelectProps = {
+  fields: {
+    inputKey: string;
+    selectKey: string;
+  };
+  options: string[];
+  defaultOption?: string;
+  className?: string;
+  placeholder?: string;
+  isReadOnly?: boolean;
+};
+
+export const NumberInputWithSelect = ({
```

Comment:
The it's an `AmountInput`!

Reply:
I prefer `NumberInputWithSelect` because its immediately clear from the name what it is. For `AmountInput` I would need to open the component and check.

That said I changed it to `AmountInput`.